sys/linux/auto.txt

```diff
@@ -6690,40 +8849,220 @@ wwan_rtnl_policy$auto_wwan_core [
 	IFLA_WWAN_LINK_ID	nlattr[IFLA_WWAN_LINK_ID, int32]
 ] [varlen]
 
+xfrm_address_filter$auto_record {
+	saddr	saddr$auto_record
+	daddr	daddr$auto_record
+	family	int16
+	splen	int8
+	dplen	int8
+}
+
+xfrm_algo$auto_record {
+	alg_name	array[int8, 64]
+	alg_key_len	int32
+	alg_key		array[int8]
+}
+
+xfrm_algo_aead$auto_record {
+	alg_name	array[int8, 64]
+	alg_key_len	int32
+	alg_icv_len	int32
+	alg_key		array[int8]
+}
+
+xfrm_algo_auth$auto_record {
+	alg_name	array[int8, 64]
+	alg_key_len	int32
+	alg_trunc_len	int32
+	alg_key		array[int8]
+}
+
+xfrm_encap_tmpl$auto_record {
+	encap_type	int16
+	encap_sport	int16
+	encap_dport	int16
+	encap_oa	encap_oa$auto_record
+}
+
+xfrm_id$auto_record {
+	daddr	daddr$auto_record
+	spi	int32
+	proto	int8
+}
+
+xfrm_lifetime_cfg$auto_record {
+	soft_byte_limit			intptr
+	hard_byte_limit			intptr
+	soft_packet_limit		intptr
+	hard_packet_limit		intptr
+	soft_add_expires_seconds	intptr
+	hard_add_expires_seconds	intptr
+	soft_use_expires_seconds	intptr
+	hard_use_expires_seconds	intptr
+}
+
+xfrm_lifetime_cur$auto_record {
+	bytes		intptr
+	packets		intptr
+	add_time	intptr
+	use_time	intptr
+}
+
 xfrm_lwt_policy$auto_xfrm_interface_core [
 	LWT_XFRM_IF_ID	nlattr[LWT_XFRM_IF_ID, int32]
 	LWT_XFRM_LINK	nlattr[LWT_XFRM_LINK, int32]
 ] [varlen]
 
+xfrm_mark$auto_record {
+	v	int32
+	m	int32
+}
+
+xfrm_replay_state$auto_record {
+	oseq	int32
+	seq	int32
+	bitmap	int32
+}
+
+xfrm_replay_state_esn$auto_record {
+	bmp_len		int32
+	oseq		int32
+	seq		int32
+	oseq_hi		int32
+	seq_hi		int32
+	replay_window	int32
+	bmp		array[int32]
+}
+
+xfrm_selector$auto_record {
+	daddr		daddr$auto_record
+	saddr		saddr$auto_record
+	dport		int16
+	dport_mask	int16
+	sport		int16
+	sport_mask	int16
+	family		int16
+	prefixlen_d	int8
+	prefixlen_s	int8
+	proto		int8
+	ifindex		int32
+	user		int32
+}
+
+xfrm_stats$auto_record {
+	replay_window		int32
+	replay			int32
+	integrity_failed	int32
+}
+
+xfrm_user_kmaddress$auto_record {
+	local		local$auto_record
+	remote		remote$auto_record
+	reserved	int32
+	family		int16
+}
+
+xfrm_user_migrate$auto_record {
+	old_daddr	old_daddr$auto_record
+	old_saddr	old_saddr$auto_record
+	new_daddr	new_daddr$auto_record
+	new_saddr	new_saddr$auto_record
+	proto		int8
+	mode		int8
+	reserved	int16
+	reqid		int32
+	old_family	int16
+	new_family	int16
+}
+
+xfrm_user_offload$auto_record {
+	ifindex	int32
```

Comment:
Nit: we can make it be of the `ifindex` type like we did in Netlink descriptions.